extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -1,43 +1,121 @@
+  Uri,
+  ProgressOptions,
+  ProgressLocation,
+  commands,
+  window,
+} from "vscode";
 import * as fs from "fs-extra";
 import * as path from "path";
 import { DatabaseManager, DatabaseItem } from "./databases";
-import { ProgressCallback, showAndLogErrorMessage, withProgress, showAndLogInformationMessage } from "./helpers";
+import {
+  ProgressCallback,
+  showAndLogErrorMessage,
+  withProgress,
+  showAndLogInformationMessage,
+} from "./helpers";
 
 /**
  * Prompts a user to fetch a database from a remote location. Database is assumed to be an archive file.
  *
  * @param databasesManager the DatabaseManager
  * @param storagePath where to store the unzipped database.
  */
-export async function promptImportInternetDatabase(databasesManager: DatabaseManager, storagePath: string): Promise<DatabaseItem | undefined> {
+export async function promptImportInternetDatabase(
+  databasesManager: DatabaseManager,
+  storagePath: string
+): Promise<DatabaseItem | undefined> {
   let item: DatabaseItem | undefined = undefined;
 
   try {
     const databaseUrl = await window.showInputBox({
-      prompt: 'Enter URL of zipfile of database to download'
+      prompt: "Enter URL of zipfile of database to download",
     });
     if (databaseUrl) {
       validateHttpsUrl(databaseUrl);
 
       const progressOptions: ProgressOptions = {
         location: ProgressLocation.Notification,
-        title: 'Adding database from URL',
+        title: "Adding database from URL",
         cancellable: false,
       };
-      await withProgress(progressOptions, async progress => (item = await databaseArchiveFetcher(databaseUrl, databasesManager, storagePath, progress)));
-      commands.executeCommand('codeQLDatabases.focus');
+      await withProgress(
+        progressOptions,
+        async (progress) =>
+          (item = await databaseArchiveFetcher(
+            databaseUrl,
+            databasesManager,
+            storagePath,
+            progress
+          ))
+      );
+      commands.executeCommand("codeQLDatabases.focus");
     }
-    showAndLogInformationMessage('Database downloaded and imported successfully.');
+    showAndLogInformationMessage(
+      "Database downloaded and imported successfully."
+    );
   } catch (e) {
     showAndLogErrorMessage(e.message);
   }
 
   return item;
 }
 
+/**
+ * Prompts a user to fetch a database from lgtm.
+ * User enters a project url and then the user is asked which language
+ * to download (if there is more than one)
+ *
+ * @param databasesManager the DatabaseManager
+ * @param storagePath where to store the unzipped database.
+ */
+export async function promptImportLgtmDatabase(
+  databasesManager: DatabaseManager,
+  storagePath: string
+): Promise<DatabaseItem | undefined> {
+  let item: DatabaseItem | undefined = undefined;
+
+  try {
+    const lgtmUrl = await window.showInputBox({
+      prompt:
+        "Enter the project URL on LGTM (e.g., https://lgtm.com/projects/g/github/codeql)",
```

Comment:
Any thoughts on why not to require simply the owner/project string, in this example, just `github/codeql`? I feel like that would save a bit of validation complexity.

Reply:
Or do we not have confidence the prefix is always going to be `https://lgtm.com/projects/g/` in the future?

Reply:
I was thinking it's just easier for a user to copy the full URL without thinking about which substring to copy and paste it here. And the other thing is that I don't really know what the `g` is.

Maybe I should first get some clarification on what that`g` means.

Reply:
The `g` means that it's a GitHub project (LGTM also has some [GitLab](https://lgtm.com/projects/gl/sj_vs/lgtm-gitlab-integration-test/) and [BitBucket](https://lgtm.com/projects/b/bitbucket/django_features/) ones.)

Reply:
Hm, that's fair. In that case I might suggest heuristically accepting
`https://lgtm.com/projects/g/github/codeql/?mode=list`
and
`https://lgtm.com/projects/g/github/codeql/alerts`
since the former is what `https://lgtm.com/projects/g/github/codeql/` seems to redirect to that, and `.../alerts` urls are suggested by the lgtm interface in places?

Dunno, not important for this PR necessarily, just a nicety since it's possible to DWIM in these cases.

Reply:
I was thinking it's just easier for a user to copy the full URL without thinking about which substring to copy and paste it here. And the other thing is that I don't really know what the `g` is.

Maybe I should first get some clarification on what that`g` means.

Reply:
Ok, so how about just checking that we're at `(www.)lgtm.com`, and there are at least (instead of exactly) 4 components to the path, and the first is `projects`, and we pull out the third and fourth?

Reply:
> Ok, so how about just checking that we're at `(www.)lgtm.com`, and there are at least (instead of exactly) 4 components to the path, and the first is `projects`, and we pull out the third and fourth?

Yes, indeed. Except that you should pull out the second, third and fourth. The URL pattern is `/projects/{provider}/{org}/{name}/sub_pages...`